src/profile/components/basicData/BasicData.tsx

```diff
@@ -1,28 +1,172 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { Field, Formik, FormikProps, Form } from 'formik';
+import { TextInput } from 'hds-react';
+import to from 'await-to-js';
 
 import commonFormStyles from '../../../common/cssHelpers/form.module.css';
 import LabeledValue from '../../../common/labeledValue/LabeledValue';
 import ProfileSection from '../../../common/profileSection/ProfileSection';
 import { getFormFields } from '../../helpers/formProperties';
-import { basicDataType, BasicDataValue } from '../../helpers/editData';
-import { useProfileDataEditor } from '../../hooks/useProfileDataEditor';
+import {
+  basicDataType,
+  BasicDataValue,
+  EditDataValue,
+} from '../../helpers/editData';
+import {
+  ActionListener,
+  saveTypeToAction,
+  useProfileDataEditor,
+} from '../../hooks/useProfileDataEditor';
+import { basicDataSchema } from '../../../common/schemas/schemas';
+import { getFieldError, getIsInvalid } from '../../helpers/formik';
+import useNotificationContent from '../editingNotifications/useNotificationContent';
+import EditingNotifications from '../editingNotifications/EditingNotifications';
+import EditButtons, { ActionHandler } from '../editButtons/EditButtons';
+import FormButtons from '../formButtons/FormButtons';
+import SaveIndicator from '../saveIndicator/SaveIndicator';
+
+type FormikValues = BasicDataValue;
 
 function BasicData(): React.ReactElement | null {
-  const { editDataList } = useProfileDataEditor({
+  const { editDataList, save, reset } = useProfileDataEditor({
     dataType: basicDataType,
   });
+  const [isEditing, setEditing] = useState(false);
   const { t } = useTranslation();
+  const {
+    content,
+    setErrorMessage,
+    setSuccessMessage,
+    clearMessage,
+  } = useNotificationContent();
   const testId = basicDataType;
 
   if (!editDataList || !editDataList[0]) {
     return null;
   }
   const editData = editDataList[0];
-  const { value } = editData;
+  const { value, saving } = editData;
   const { firstName, nickname, lastName } = value as BasicDataValue;
   const formFields = getFormFields(basicDataType);
 
+  const hasFieldError = (
+    formikProps: FormikProps<FormikValues>,
+    fieldName: keyof FormikValues
+  ): boolean => getIsInvalid<FormikValues>(formikProps, fieldName, true);
+
+  const getFieldErrorMessage = (
+    formikProps: FormikProps<FormikValues>,
+    fieldName: keyof FormikValues
+  ) => {
```

Comment:
This seems to be missing typings for the return value?

Reply:
TS compiler / linter and Sonar would alert if needed. TS understands the return type is the return type of the  function called: "return getFieldError(....)"